src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```
@@ -61,4 +71,83 @@ class WorkspaceService(protected val argUserInfo: UserInfo, val rawlsDAO: RawlsD
     }
   }
 
+  def exportWorkspaceAttributes(workspaceNamespace: String, workspaceName: String, filename: String): Future[PerRequestMessage] = {
+    Try(rawlsDAO.getWorkspace(workspaceNamespace, workspaceName)) match {
+      case Failure(regret) => Future(RequestCompleteWithErrorReport(StatusCodes.BadRequest, regret.getMessage))
```

2 comments: 1) you don't need a full fledged Future for something we already know the value of. In this case Future.successful creates something that looks like a Future but completes immediately. 2) Why is this a bad request? Seems like you are masking the underlying error. Maybe it was a 404? Maybe it was a communications error?

---

why do you even need to handle the error here at all? Is there something special about the error handling that is different than what per request would do?

---

and 1 more: this is probably not doing what you think it is. The Try is not wrapping the actual call to rawls. It is wrapping only the construction of the call. rawlsDAO.getWorkspace(workspaceNamespace, workspaceName) creates a Future, the Try wraps the creation of the Future, not the actual execution. So calling this error a bad request is wrong, it is an internal server error.

TLDR; get rid of the Try